index.ts

```diff
@@ -113,76 +98,213 @@ function isInSnapZone() {
-        if (!vibrancyOp.targetOpacity) vibrancyOp.targetOpacity = vibrancyOp.colors.a
-    } catch (e) {
-    }
+    if (options.effect === 'acrylic' && isRS4OrGreater())
+        config.effect = 1;
 
     // Debug output
-    if (_vibrancyDebug) console.log(vibrancyOp)
+    if (_vibrancyDebug)
+        console.log(config)
+
+    return config;
+}
+
+/**
+ * The theme to apply to the vibrancy. Can be 'light',
+ * 'dark', 'appearance-based' or a custom HEX color
+ * with alpha.
+ */
+type VibrancyTheme = 'light' | 'dark' | 'appearance-based' | { hex: string } | { rgba: [number, number, number, number] };
+
+/**
+ * The effect to apply. Can be 'acrylic' or 'blur'.
+ */
+type VibrancyEffect = 'acrylic' | 'blur';
+
+/**
+ * The vibrancy object
+ */
+interface VibrancyOptions {
+    /**
+     * The theme to use.
+     */
+    theme?: VibrancyTheme;
+
+    /**
+     * The effect to use.
+     */
+    effect?: VibrancyEffect;
+
+    /**
+     * If enabled, we use a custom window resize/move
+     * handler for performance.
+     */
+    useCustomWindowRefreshMethod?: boolean;
+
+    /**
+     * Maximum value to refresh application screen
+     * in seconds.
+     */
+    maximumRefreshRate?: number;
+
+    /**
+     * If true, acrylic effect will be disabled whe
+     * window lost focus.
+     */
+    disableOnBlur?: boolean;
+
+    debug?: boolean
+}
 
-    return vibrancyOp;
+type Vibrancy = VibrancyTheme | VibrancyOptions
+
+/**
+ * Allow modifying default BrowserWindowConstructorOptions
+ * to change vibrancy to VibrancyOptions.
+ */
+type Modify<T, R> = Omit<T, keyof R> & R;
+
+/**
+ * The new options of the BrowserWindow with the VibrancyOptions.
+ */
+type AcrylicBrowserWindowConstructorOptions = Modify<electron.BrowserWindowConstructorOptions, {
+
+    /**
+     * The vibrancy settings for the window. Can be
+     * a VibrancyTheme or the VibrancyOptions object.
+     */
+    vibrancy?: Vibrancy
+}>;
+
+interface WindowConfig {
+    vibrancyActivated: boolean
+    vibrnacyConfig: VibrancyConfig
 }
 
-class vBrowserWindow extends eBrowserWindow {
-    constructor(props) {
-        let oShow = props.show;
-        if (!('show' in props)) oShow = true;
-        if (props.vibrancy && props.vibrancy.debug) _vibrancyDebug = true;
-        let vibrancyOp = opFormatter(props.vibrancy);
-        if (isWindows10() && vibrancyOp) {
+const windowConfigs: Record<number, WindowConfig> = {}
+
+class BrowserWindow extends electron.BrowserWindow {    
+    constructor(options?: AcrylicBrowserWindowConstructorOptions) {
+        super(Object.assign(options, { vibrancy: undefined }))
+
+        let oShow = options?.show ?? true;
+
+        if (typeof options?.vibrancy === 'object' && 'debug' in options.vibrancy && options.vibrancy.debug)
+            _vibrancyDebug = true;
+
+
+        let config = opFormatter(options?.vibrancy);
+
+        if (isWindows10() && config) {
             props.vibrancy = null;
```

Comment:
What does this line of code do? There are no references of `vibrancy` after this. Why is `vibrancy` set to null only if Windows 10?

Reply:
I explicitly set vibrancy to null because unwanted behavior may occur when creating BrowserWindow objects below.

Reply:
Should that apply to all platforms?

Reply:
No, the current code is correct.

Reply:
The macOS version of the electron can have vibrancy effect without using this module by inserting the vibrancy option in the browswindow generator. This code allows you to ignore vibrancy-related options in the windows from the original browser window constructor, but call the generator in macOS as the vibrancy option is.